### PR TITLE
Feature Add redux

### PR DESCRIPTION
**Pull request by kelvinbush**
In this pull request I did the following:
- Added redux dependencies.
- Add actions for books and categories.
- Added reducers for books and categories.
- Added a root reducer using `combineReducers`.
- Added the store to the react app.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const ADD_BOOK = 'ADD_BOOK';
+const DELETE_BOOK = 'DELETE_BOOK';
```
Comment (Kingstalux) [inline src/redux/books/books.js]: - [ ] Good job so far 👍 Kindly make sure that the values of your action type follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Therefore it should be something like this: `const ADD_BOOK = bookstore/books/ADD_BOOK`. Kindly rectify this 💯

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (Kingstalux) [inline src/redux/categories/categories.js]: - [x] Kindly use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for this action type 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @kelvinbush, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
149637855-7722d04c-b45d-4185-925f-7de047106461


## Highlights 👏 
✅ Good PR description 🔥 
✅ Professional README file 💯 
✅ Good commit history 👍 
✅ Linters are passing 👌 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by smunozmo — APPROVED** [review]
Hi @kelvinbush, 👋 

## Highlights 👏 

- Reducers and actions are set up correctly, good job!
- You are using the duck pattern correctly, great!
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

### APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**